tests/factories.py

```diff
@@ -12,109 +13,120 @@
-    name = factory.Sequence(lambda n: 'Organization ' + str(n))
-    slug = factory.Sequence(lambda n: 'org_' + str(n))
+class OrganizationFactory(DjangoModelFactory):
+    name = Sequence(lambda n: 'Organization ' + str(n))
+    slug = Sequence(lambda n: 'org_' + str(n))
 
     class Meta:
         model = organization_models.Organization
         django_get_or_create = ['name']
 
 
-class FacilityFactory(factory.DjangoModelFactory):
-    name = factory.Sequence(lambda n: 'Facility ' + str(n))
-    slug = factory.Sequence(lambda n: 'facility_' + str(n))
+class FacilityFactory(DjangoModelFactory):
+    name = Sequence(lambda n: 'Facility ' + str(n))
+    slug = Sequence(lambda n: 'facility_' + str(n))
 
-    place = factory.SubFactory(PlaceFactory)
-    organization = factory.SubFactory(OrganizationFactory)
+    place = SubFactory(PlaceFactory)
+    organization = SubFactory(OrganizationFactory)
 
     class Meta:
         model = organization_models.Facility
         django_get_or_create = ['name', 'place', 'organization']
 
 
-class TaskFactory(factory.DjangoModelFactory):
+class TaskFactory(DjangoModelFactory):
     class Meta:
         model = organization_models.Task
 
-    name = factory.Sequence(lambda n: 'Task ' + str(n))
-    description = factory.Sequence(lambda n: 'task ' + str(n))
+    name = Sequence(lambda n: 'Task ' + str(n))
+    description = Sequence(lambda n: 'task ' + str(n))
 
-    facility = factory.SubFactory(FacilityFactory)
+    facility = SubFactory(FacilityFactory)
 
 
-class ShiftFactory(factory.DjangoModelFactory):
+class WorkplaceFactory(DjangoModelFactory):
+    class Meta:
+        model = organization_models.Workplace
+
+    name = Sequence(lambda n: 'Workplace ' + str(n))
+    description = Sequence(lambda n: f'workplace {n}')
+
+    facility = SubFactory(FacilityFactory)
+
+
+class ShiftFactory(DjangoModelFactory):
     class Meta:
         model = scheduler_models.Shift
 
-    task = factory.SubFactory(TaskFactory)
-    facility = factory.SubFactory(FacilityFactory)
+    task = SubFactory(TaskFactory)
+    facility = SubFactory(FacilityFactory)
+    workplace = SubFactory(WorkplaceFactory)
 
     starting_time = datetime(2016, 2, 13, 19, 0)
     ending_time = datetime(2016, 2, 13, 20, 0)
     slots = 10
 
 
-class UserFactory(factory.DjangoModelFactory):
+class UserFactory(DjangoModelFactory):
     class Meta:
         model = User
 
     username = FuzzyText(length=10, chars=string.ascii_letters, prefix='')
     first_name = FuzzyText(length=10, chars=string.ascii_letters, prefix='')
     last_name = FuzzyText(length=10, chars=string.ascii_letters, prefix='')
-    password = factory.PostGenerationMethodCall('set_password', 'defaultpassword')
-    email = factory.LazyAttribute(lambda o: '%s@example.com' % o.last_name)
+    password = PostGenerationMethodCall('set_password', 'defaultpassword')
+    email = LazyAttribute(lambda o: '%s@example.com' % o.last_name)
```

Comment:
Replace with an f-string?

```suggestion
    email = LazyAttribute(lambda o: f"{o.last_name}@example.com")
```

Reply:
*meeehhhh* didn't touch too much code, but: valid point.